security/pkg/util/certutil.go

```diff
@@ -51,8 +53,10 @@ func (cu CertUtilImpl) GetWaitTime(certBytes []byte, now time.Time, minGracePeri
 		return time.Duration(0), fmt.Errorf("certificate already expired at %s, but now is %s",
 			cert.NotAfter, now)
 	}
-	gracePeriod := cert.NotAfter.Sub(cert.NotBefore) * time.Duration(cu.gracePeriodPercentage) / time.Duration(100)
+	gracePeriod := time.Duration(float64(cert.NotAfter.Sub(cert.NotBefore)) * (float64(cu.gracePeriodPercentage) / 100))
```

Comment:
Thanks for catching this issue. Multiply time.Duration(int64) by a int could cause overflow.

Reply:
A comment has been added to explain the overflow.